src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -129,7 +125,43 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
             class="more-info"
             @click="${this._handleMoreInfo}"
           ></paper-icon-button>
-          <div id="thermostat"></div>
+          <div id="thermostat">
+            ${stateObj.state === "unavailable"
+              ? html`
+                  <round-slider
+                    .radius=${this._radius}
+                    disabled="true"
+                  ></round-slider>
+                `
+              : stateObj.attributes.target_temp_low &&
+                stateObj.attributes.target_temp_high
+              ? html`
+                  <round-slider
+                    .radius=${this._radius}
+                    .low=${stateObj.attributes.target_temp_low}
+                    .high=${stateObj.attributes.target_temp_high}
+                    .min=${stateObj.attributes.min_temp}
+                    .max=${stateObj.attributes.max_temp}
+                    .step=${this._stepSize}
+                    @value-changing=${this._dragEvent}
+                    @value-changed=${this._setTemperature}
+                  ></round-slider>
+                `
+              : html`
+                  <round-slider
+                    .radius=${this._radius}
+                    .value=${stateObj.attributes.temperature !== null &&
+                    Number.isFinite(Number(stateObj.attributes.temperature))
+                      ? stateObj.attributes.temperature
+                      : stateObj.attributes.min_temp}
+                    .step=${this._stepSize}
+                    .min=${stateObj.attributes.min_temp}
+                    .max=${stateObj.attributes.max_temp}
+                    @value-changing=${this._dragEvent}
+                    @value-changed=${this._setTemperature}
+                  ></round-slider>
```

Comment:
Actually, I think round-slider could handle this. If you give it both low, high and value, it will show one handle if low is undefined, and two otherwise.
May be a bit confusing, though...